CONTRIBUTUNG.md

```diff
@@ -0,0 +1,44 @@
+Please always report problems by filling an issue here on GitHub. This also covers reporting problems with the documentation (e.g. if sth is unclear).
+
+General rules:
+
+    1. the code style suggestions of PyCharm should be followed
```

Comment:
I would not formulate this in terms of PyCharm. I am not using PyCharm and I dont want to start. @albertz do you know if we can have the same code suggestions as a cmd-line tool?

Reply:
Can we not just add a check for "black" as a github PR hook?

Reply:
Black is just a formater, and by far does not cover all what PEP8 implies. (E.g. I'm sure it will not rename your variables.)

We actually use PyCharm as command line tool for automatic testing of code style in RETURNN, as part of CI, via [this script](https://github.com/rwth-i6/returnn/blob/master/tests/pycharm-inspect.py).

As already mentioned, part of that is `pycodestyle`, which can be used as a separate tool. (PyCharm by default ignores `E121,E123,E126,E226,E24,E704,W503,W504` which it probably checks in better ways - I haven't really looked into that.)

PyCharm still checks some more things, which are all reasonable. I don't think there are any other tools which can do the same. But we can of course check those also my human reviews. (All of the things are very reasonable anyway.)

There is also pylint. I tried to set it up once, i.e. to come up with reasonable defaults, but I gave up at some point because it misses many important things which are all covered by PyCharm, and also had some false positives, and otherwise some strange defaults.

Reply:
> Can we not just add a check for "black" as a github PR hook?

Right, black should always be part of the pipeline. The other PEP8 things are added on top of that.

Reply:
I also forgot MyPy, which can do type checks (but also just like PyCharm).